test/e2e/client_tls_test.go

```diff
@@ -384,6 +313,535 @@ func TestClientTLS(t *testing.T) {
+				t.Fatalf("failed to observe expected conditions: %v", err)
+			}
+
+			clientCertsConfigmap := corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "client-certificates",
+					Namespace: namespace.Name,
+				},
+				Data: map[string]string{},
+			}
+			for name, keyCert := range tcCerts.ClientCerts.Accepted {
+				clientCertsConfigmap.Data[name+".key"] = encodeKey(keyCert.Key)
+				clientCertsConfigmap.Data[name+".pem"] = keyCert.CertPem
+			}
+			for name, keyCert := range tcCerts.ClientCerts.Rejected {
+				clientCertsConfigmap.Data[name+".key"] = encodeKey(keyCert.Key)
+				clientCertsConfigmap.Data[name+".pem"] = keyCert.CertPem
+			}
+			if err := kclient.Create(context.TODO(), &clientCertsConfigmap); err != nil {
+				t.Fatalf("failed to create configmap %q: %v", clientCertsConfigmap.Name, err)
+			}
+			defer assertDeleted(t, kclient, &clientCertsConfigmap)
+
+			// Use the router image for the exec pod since it has curl.
+			routerDeployment := &appsv1.Deployment{}
+			routerDeploymentName := controller.RouterDeploymentName(ic)
+			if err := kclient.Get(context.TODO(), routerDeploymentName, routerDeployment); err != nil {
+				t.Fatalf("failed to get routerDeployment %q: %v", routerDeploymentName, err)
+			}
+
+			podName := "mtls-with-crls-client"
+			image := routerDeployment.Spec.Template.Spec.Containers[0].Image
+			clientPod := buildExecPod(podName, namespace.Name, image)
+			clientPod.Spec.Volumes = []corev1.Volume{{
+				Name: "client-certificates",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: clientCertsConfigmap.Name,
+						},
+					},
+				},
+			}}
+			clientPod.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{{
+				Name:      "client-certificates",
+				MountPath: "/tmp/tls/",
+				ReadOnly:  true,
+			}}
+			clientPodName := types.NamespacedName{
+				Name:      clientPod.Name,
+				Namespace: clientPod.Namespace,
+			}
+			if err := kclient.Create(context.TODO(), clientPod); err != nil {
+				t.Fatalf("failed to create pod %q: %v", clientPodName, err)
+			}
+			defer assertDeleted(t, kclient, clientPod)
+
+			err = wait.PollImmediate(2*time.Second, 3*time.Minute, func() (bool, error) {
+				if err := kclient.Get(context.TODO(), clientPodName, clientPod); err != nil {
+					t.Logf("failed to get client pod %q: %v", clientPodName, err)
+					return false, nil
+				}
+				for _, cond := range clientPod.Status.Conditions {
+					if cond.Type == corev1.PodReady {
+						return cond.Status == corev1.ConditionTrue, nil
+					}
+				}
+				return false, nil
+			})
+			if err != nil {
+				t.Fatalf("timed out waiting for pod %q to become ready: %v", clientPodName, err)
+			}
+
+			// We need an IP address to which to send requests.  The test client
```

Comment:
*nit* I struggled a bit on this comment. But I realized it's because the curl function doesn't directly use the DNS name. Is there a better way we can describe this?

```suggestion
			// We need an IP address to which to send requests (in lieu of using a DNS domain name). The test client
```